### ISSUE TITLE
dcgm[-exporter] should detect crashed/hung GPU's not be dependent CLI tools

### ISSUE THREAD

**Issue by maxx** [NVIDIA/gpu-monitoring-tools#46]
dcgm-exporter on a node with hung GPU's results in a hung call to nvidia-smi in this awk mashup:

https://github.com/NVIDIA/gpu-monitoring-tools/blob/master/exporters/prometheus-dcgm/dcgm-exporter/dcgm-exporter#L51-L62

The only avertable/actionable thing in this scenario is stale metrics in node exporter, which are not immediate, or necessarily reliable.

Additionally, sometimes nvidia-smi exits immediately in a crashed GPU scenario indicating a GPU "feel off the bus", and the host requires reboot.  In this scenario, nvidia-smi does not return any data to dcgm-exporter, and it simply stops reporting metrics.

dcgm-exporter should be GPU fault tolerant and also expose GPU health status.

**Comment by nvjmayo** [comment]
in your hung call is nvidia-smi unkillable (stuck in 'D' state) ? if so this means the driver is hung and it is difficult for us to query GPU at a high level, as the driver is responsible for setting up channels (nvrm)

I am marking this issue as a feature request. I think a lighter weight query that perhaps only checks driver health is needed before a more robust status check is possible.

**Comment by maxx** [comment]
That's right -- it's generally in 'D' state.  It's been tricky to set up an alert which corresponds with dcgm-exporter hitting a hung nvidia-smi, so what we've had to do as a workaround is roll our own node exporter textfile which then exports nvidia-smi runtime and return code every 5 minutes via a systemd timer. While dcgm-exporter provides a lot of useful information when things are working well, this workaround has been the only reliable way to find hung GPU's which in some cases can blackhole jobs in kubernetes.
